packages/calypso-e2e/src/lib/pages/my-home-page.ts

```diff
@@ -68,9 +67,13 @@ export class MyHomePage {
 	 * @returns {string} No return value.
 	 */
 	async suggestedDomainName(): Promise< string > {
-		await this.page.locator( selectors.domainUpsellDomainAvailable ).waitFor();
+		await this.page.locator( selectors.domainUpsellSuggestedDomain ).waitFor();
 		const elementHandle = await this.page.waitForSelector( selectors.domainUpsellSuggestedDomain );
-		return await elementHandle.innerText();
+		const domainName = await elementHandle.textContent();
+		if ( domainName === null ) {
+			throw new Error( 'Domain name is null' );
+		}
+		return domainName;
```

Comment:
**suggestion** - there's a way to do this process of 1) checking whether a suggestion card is present, then 2) extracting the suggested domain out of that card, without using discouraged APIs.

GitHub doesn't allow me to select the entire method to make a suggestion, but the following is one example:

```suggestion
		const locator = this.page.locator( '.domain-upsell__card');
		
		await locator.waitFor();
		
		return await locator.locator('text').textContent();
```

Note that the suggested snippet sidesteps the use of `ElementHandle` API, [which has been discouraged by Playwright maintainers for some time](https://playwright.dev/docs/api/class-elementhandle). 

I've further changed the method to return whatever the text content is contained in in the target element, opting to leave the validation up to the caller, like this:

```typescript
// In a test step
const suggestedDomain = await myHomePage.suggestedDomain();
expect( suggestedDomain ).toBeTruthy();
```

Anyway, both are merely suggestions, but I would highly recommend the first suggestion to remove the use of ElementHandler API.